--- a/pam_yubico.c
+++ b/pam_yubico.c
@@ -906,6 +906,13 @@ pam_sm_authenticate (pam_handle_t * pamh,
 
   DBG (("OTP: %s ID: %s ", otp, otp_id));
 
+  /* Verify the OTP */
+  if (check_modhex(otp, strlen(otp))) {
+     DBG (("OTP contains non-authorized character (non modhex): '%s'.", otp));
+     retval = PAM_AUTH_ERR;
+     goto done;
+  }
+
   /* user entered their system password followed by generated OTP? */
   if (password_len > TOKEN_OTP_LEN + cfg->token_id_length)
     {
--- a/util.c
+++ b/util.c
@@ -332,3 +332,28 @@ write_chalresp_state(FILE *f, CR_STATE *state)
   return 0;
 }
 #endif /* HAVE_CR */
+
+int
+check_modhex(const char *input, const int len)
+{
+  const char *end;
+  for (end = input + len; input < end; ++input) {
+    if (*input < 0x62) {
+      return -1;
+    }
+    if (*input > 0x76) {
+      return -1;
+    }
+    if ((*input > 0x6C)
+        && (*input < 0x74)) {
+      switch (*input) {
+        case 'n':
+        case 'r':
+          break;
+        default:
+          return -1;
+      }
+    }
+  }
+  return 0;
+}
--- a/util.h
+++ b/util.h
@@ -55,6 +55,7 @@
 #endif /* DEBUG_PAM */
 
 int get_user_cfgfile_path(const char *common_path, const char *filename, const char *username, char **fn);
+int check_modhex(const char *input, const int len);
 
 #if HAVE_CR
 
-- 
1.7.3.4

