--- anna.cpp	2011-01-18 11:39:10.000000000 +0100
+++ anna.cpp	2011-06-07 13:59:55.784228582 +0200
@@ -381,30 +381,32 @@
 		}
 		else
 		{
-			if (hard_only)
-			{
-				if (prev[prev_index].state_type == ST_HARD && cur[loop].state_type == ST_HARD)
+			if (!(prev[prev_index].is_flapping && cur[loop].is_flapping)) {
+				if (hard_only)
 				{
-					if (prev[prev_index].current_state != cur[loop].current_state)
+					if (prev[prev_index].state_type == ST_HARD && cur[loop].state_type == ST_HARD)
 					{
-						show = 1;
+						if (prev[prev_index].current_state != cur[loop].current_state)
+						{
+							show = 1;
+						}
+					}
+					else if (prev[prev_index].state_type == ST_SOFT && cur[loop].state_type == ST_HARD)
+					{
+						if (cur[loop].current_state != 0)
+						{
+							show = 1;
+						}
 					}
 				}
-				else if (prev[prev_index].state_type == ST_SOFT && cur[loop].state_type == ST_HARD)
+				else
 				{
-					if (cur[loop].current_state != 0)
+					if (prev[prev_index].current_state != cur[loop].current_state)
 					{
 						show = 1;
 					}
 				}
 			}
-			else
-			{
-				if (prev[prev_index].current_state != cur[loop].current_state)
-				{
-					show = 1;
-				}
-			}
 		}
 
 		/* also 'unknown' is emitted when 'critical_only' is selected */
