--- include/config.h.orig       2009-09-15 18:02:38.000000000 +0200
+++ include/config.h    2009-09-15 18:02:57.000000000 +0200
@@ -1065,9 +1065,9 @@
 #define IRCD_PIDFILE PPATH
 
 /* enforce a minimum, even though it'll probably break at runtime */
-#if (MAXCONNECTIONS < 20)
+#if (MAXCONNECTIONS < 100)
 # undef MAXCONNECTIONS
-# define MAXCONNECTIONS 20
+# define MAXCONNECTIONS 100
 #endif
 
 #if (MAXCONNECTIONS > 1000)
